pages/evenements-precedents/[year].tsx

```diff
@@ -0,0 +1,77 @@
+import { GetStaticPaths, GetStaticProps, NextPage } from 'next';
+import React from 'react';
+import { LyonJSHead } from '../../modules/header/LyonJSHead';
+import { Event } from '../../modules/event/types';
+import { dataOverride } from '../../data/data-override';
+import _merge from 'lodash/merge';
+import _uniq from 'lodash/uniq';
+import { fetchMeetupEvents } from '../../modules/meetup/api';
+import { ParsedUrlQuery } from 'querystring';
+import { H1 } from '../../modules/atoms/remark/titles';
+import { YearNavigation } from '../../modules/event/YearNaviation';
+import { EventTile } from '../../modules/event/EventTile';
+import Link from 'next/link';
+
+const Event: NextPage<{ pastEvents: Event[]; years: string[]; year: string }> = ({ pastEvents, years, year }) => {
+  return (
+    <>
+      <LyonJSHead title={`LyonJS | Évènements ${year}`} />
+      <main>
+        <H1>Évènements</H1>
+
+        <YearNavigation years={years} />
+
+        {pastEvents.map((event) => (
+          <Link key={event.eventUrl} href={event.eventUrl}>
+            <EventTile event={event} />
+          </Link>
+        ))}
+      </main>
+    </>
+  );
+};
+
+const overrideEvent = (event: Event): Event => {
+  if (event && dataOverride[event.eventUrl]) {
+    return _merge(event, dataOverride[event.eventUrl]);
+  }
+  return event;
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const { pastEvents } = await fetchMeetupEvents();
+  const yearsFromEvents: string[] = pastEvents
+    .map((event) => new Date(event.dateTime).getFullYear())
+    .map((year) => year.toString());
+  const years = _uniq(yearsFromEvents);
+
+  return {
+    paths: years.map((year) => ({ params: { year } })),
+    fallback: false,
+  };
+};
+
+interface Params extends ParsedUrlQuery {
+  year: string;
+}
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const { pastEvents } = await fetchMeetupEvents();
```

Comment:
Do you know why there are 3 API requests (2 GET and 1 HEAD) to the same json each time we navigate to a different year ? 🤔

Reply:
No idea for now, as the page is statically generated in production, we won't have issue because we won't call meetup API.